components/connectivity-adapter/internal/appregistry/service/handler_test.go

```diff
@@ -0,0 +1,116 @@
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	gcli "github.com/machinebox/graphql"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/gorilla/mux"
+	"github.com/kyma-incubator/compass/components/connectivity-adapter/pkg/gqlcli/automock"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const url = "http://doesnt.really/matter"
+
+func TestHandler_Delete(t *testing.T) {
+	id := "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
+	httpReq := fixRequest(id, strings.NewReader(""))
+	expectedGQLReq := prepareUnregisterApplicationRequest(id)
+	notFoundErr := fmt.Errorf("graphql: while getting Application with ID %s: Object was not found", id)
+	internalErr := errors.New("Post http://127.0.0.1:3000/graphql: dial tcp 127.0.0.1:3000: connect: connection refused")
+
+	testCases := []struct {
+		Name                       string
+		GraphQLClientErr           error
+		LoggerAssertionsFn         func(t *testing.T, hook *test.Hook)
+		ExpectedResponseStatusCode int
+		ExpectedResponseBody       string
+	}{
+		{
+			Name:                       "Success",
+			GraphQLClientErr:           nil,
+			ExpectedResponseBody:       "",
+			ExpectedResponseStatusCode: http.StatusNoContent,
+		},
+		{
+			Name:                       "Not found",
+			GraphQLClientErr:           notFoundErr,
+			ExpectedResponseBody:       fmt.Sprintf("{\"code\":2,\"error\":\"entity with ID %s not found\"}\n", id),
+			ExpectedResponseStatusCode: http.StatusNotFound,
+		},
+		{
+			Name: "Error",
+			LoggerAssertionsFn: func(t *testing.T, hook *test.Hook) {
+				assert.Equal(t, 1, len(hook.AllEntries()))
+				entry := hook.LastEntry()
+				require.NotNil(t, entry)
+				assert.Equal(t, log.ErrorLevel, entry.Level)
+				assert.Equal(t, fmt.Sprintf("while deleting service with ID %s: %s", id, internalErr.Error()), entry.Message)
+			},
+			GraphQLClientErr:           internalErr,
+			ExpectedResponseBody:       fmt.Sprintf("{\"code\":1,\"error\":\"%s\"}\n", internalErr.Error()),
+			ExpectedResponseStatusCode: http.StatusInternalServerError,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			logger, hook := test.NewNullLogger()
+
+			cli, cliProvider := fixClientAndProviderMocks(httpReq, expectedGQLReq, tc.GraphQLClientErr)
+			defer mock.AssertExpectationsForObjects(t, cli, cliProvider)
+
+			w := httptest.NewRecorder()
+
+			handler := NewHandler(cliProvider, logger)
+
+			handler.Delete(w, httpReq)
+
+			resp := w.Result()
+			defer closeRequestBody(t, resp)
+
+			bodyBytes, err := ioutil.ReadAll(resp.Body)
+			require.NoError(t, err)
+
+			assert.Equal(t, tc.ExpectedResponseStatusCode, resp.StatusCode)
+			assert.Equal(t, tc.ExpectedResponseBody, string(bodyBytes))
+
+			if tc.LoggerAssertionsFn != nil {
+				tc.LoggerAssertionsFn(t, hook)
+			}
+		})
+	}
+}
+
+func fixRequest(id string, body io.Reader) *http.Request {
```

Comment:
Rename/refactor this function or move it inside the `TestHandler_Delete`. Right now when you add another test case for different HTTP methods or URL vars, this function cannot be reused.

Reply:
Hmm, techincally it can be used for every request, because we rely on gorilla/mux, when it comes to routing - that's why the request method and URL can be anything, only the body and variable counts here. Then, this function could be used for all methods for endpoint `/services/{serviceId}`. I will rename it to `fixServiceDetailsRequest`. Is that OK for you?

Reply:
Not really, you hardcoded the `http.MethodDelete` inside.

Reply:
I changed the method to `POST`. As I wrote before, it doesn't matter, the method can be anything as we don't use it.

Reply:
As discussed, I provided there `Anything` method.